keylime/web_util.py

```diff
@@ -111,7 +111,12 @@ def init_tls_dir(component, logger=None):
             if logger:
                 logger.debug("Existing server certificate and key found in %s, not generating a new ones", tls_dir)
         else:
-            ca_util.cmd_mkcert(tls_dir, "server", password=password)
+            server_key_pw = config.get(component, "server_key_password")
+            if not server_key_pw:
+                server_key_pw = None
```

Comment:
This confuses me. If it is evaluated as False is because is `None`, `""` or `0` and you set as `None`, it is necessary or is the logic reverted and you want:

```python
if server_key_pw:
    server_key_pw = str(server_key_pw)
```

Reply:
This is to avoid the empty string `""` as the underlying code treat `None` differently from `""`. The goal is to not encrypt the key with a password if the provided password is the empty string.

Reply:
Ah, I think you are right and this is not needed anymore as the underlying code was also modified.
Let me check if it is true.

Reply:
Applied your suggestion, thanks!